docs/proposals/extensions.md

```diff
@@ -0,0 +1,88 @@
+## Proposal for extending an OpenCompose file with another OpenCompose file
+
+##### Scenario:
+
+Let's say, we have a sample OpenCompose file defining a Wordpress application -
+
+_wordpress-opencompose.yml_ -
+```yaml
+version: '0.1-dev'
+
+services:
+- name: database
+  containers:
+  - image: mariadb:10
+    env:
+    - name: MYSQL_ROOT_PASSWORD
+      value: rootpasswd
+    - name: MYSQL_DATABASE
+      value: wordpress
+    - name: MYSQL_USER
+      value: wordpress
+    - name: MYSQL_PASSWORD
+      value: wordpress
+    ports:
+    - port: 3306
+
+- name: web
+  containers:
+  - image: wordpress:4
+    env:
+    - name: WORDPRESS_DB_HOST
+      value: database:3306
+    - name: WORDPRESS_DB_PASSWORD
+      value: wordpress
+    - name: WORDPRESS_DB_USER
+      value: wordpress
+    - name: WORDPRESS_DB_NAME
+      value: wordpress
+    ports:
+    - port: 80
+      type: external
+```
+
+This file works fine for a developer, but since there is no persistent volumes defined here, we need to alter this for production use cases; but changing the original file disrupts the developer workflow.
+
+So, the production folks create another OpenCompose file which will extend _wordpress-opencompose.yml_ which will look like -
+
+_wordpress.extension.yml_
+
+```yaml
+version: '0.1-dev'
+extends: wordpress-opencompose.yml
```

Comment:
> So, how do we keep track that which extension file extends what?

I don't  think that you need to do that.



> Also, since all the files are being passed in the same -f flag, then if we want one extension file to extend multiple OpenCompose files, how do we do that?

You have service names as a unique identifier. There can't be two services with the same name even in two files if those files are passed together to opencompose tool.